tests/functional/test_config.py

```diff
@@ -55,6 +55,33 @@ def test_ethereum_network_configs(config, temp_config):
         assert actual.rinkeby.block_time == 15
 
 
+def test_network_gas_limit_default(config):
+    eth_config = config.get_config("ethereum")
+
+    assert eth_config.rinkeby.gas_limit == "auto"
+    assert eth_config.local.gas_limit == "auto"
+
+
+@pytest.mark.parametrize("gas_limit", (1234, "1234"))
+def test_network_gas_limit_config(gas_limit, config, temp_config):
+    eth_config = {
+        "ethereum": {
+            "rinkeby": {
+                "default_provider": "test",
+                "gas_limit": gas_limit,
+            }
+        }
+    }
+
+    with temp_config(eth_config):
+        actual = config.get_config("ethereum")
+
+        assert actual.rinkeby.gas_limit == "1234"
```

Comment:
I am a little surprised it is a str here and trying to figure out why that is

Reply:
Seems to be the `Union[str, int]` type declaration, going to take a look at fixing that via https://pydantic-docs.helpmanual.io/usage/model_config/#smart-union

Reply:
Updated in 4c502af781eca05c73bfa7299393040f58d2038f to fix the type and add a validator